packages/headless/src/features/commerce/search/search-slice.ts

```diff
@@ -55,10 +63,12 @@ function handlePending(state: CommerceSearchState, requestId: string) {
 
 function handleFulfilled(
   state: CommerceSearchState,
-  response: CommerceSuccessResponse
+  response: CommerceSuccessResponse,
+  query?: string
 ) {
   state.error = null;
   state.facets = response.facets;
   state.responseId = response.responseId;
   state.isLoading = false;
+  state.queryExecuted = query ?? '';
```

Comment:
Are we sure query can be undefined, esp since #3912

Reply:
It's not in relation with #3912 but because the return interface from the thunk ( `QueryCommerceAPIThunkReturn`) is defined as `queryExecuted?: string`. 

I think it would be better to force it defined always. Ill try to do the change.

Reply:
Update: Splitting in 2 interfaces introduce a lot more complexity in many reducers, apparent by a slew of UT starting to fail with the change.

Seems better to just use one interface for both types of actions, and make the query optional on the return type.